test/template_projects/c.py

```diff
@@ -0,0 +1,100 @@
+import jinja2
+from .base import TemplateProject
+
+
+spam_c_template = r'''
+#include <Python.h>
+
+{{ spam_c_top_level_add }}
+
+static PyObject *
+spam_system(PyObject *self, PyObject *args)
+{
+    const char *command;
+    int sts;
+
+    if (!PyArg_ParseTuple(args, "s", &command))
+        return NULL;
+
+    sts = system(command);
+
+    {{ spam_c_function_add | indent(4) }}
+
+    return PyLong_FromLong(sts);
+}
+
+/* Module initialization */
+
+#if PY_MAJOR_VERSION >= 3
+    #define MOD_INIT(name) PyMODINIT_FUNC PyInit_##name(void)
+    #define MOD_DEF(m, name, doc, methods, module_state_size) \
+        static struct PyModuleDef moduledef = { \
+            PyModuleDef_HEAD_INIT, name, doc, module_state_size, methods, }; \
+        m = PyModule_Create(&moduledef);
+    #define MOD_RETURN(m) return m;
+#else
+    #define MOD_INIT(name) PyMODINIT_FUNC init##name(void)
+    #define MOD_DEF(m, name, doc, methods, module_state_size) \
+        m = Py_InitModule3(name, methods, doc);
+    #define MOD_RETURN(m) return;
+#endif
+
+static PyMethodDef module_methods[] = {
+    {"system", (PyCFunction)spam_system, METH_VARARGS,
+     "Execute a shell command."},
+    {NULL}  /* Sentinel */
+};
+
+MOD_INIT(spam)
+{
+    PyObject* m;
+
+    MOD_DEF(m,
+            "spam",
+            "Example module",
+            module_methods,
+            -1)
+
+    MOD_RETURN(m)
+}
+'''
+
+setup_py_template = r'''
+from setuptools import setup, Extension
+
+{{ setup_py_add }}
+
+setup(
+    ext_modules=[Extension('spam', sources=['spam.c'])],
+    {{ setup_py_setup_args_add | indent(4) }}
+)
+'''
+
+setup_cfg_template = r'''
+[metadata]
+name = spam
+version = 0.1.0
+
+{{ setup_cfg_add }}
+'''
+
+
+class CTemplateProject(TemplateProject):
+    def __init__(self, *, spam_c_top_level_add='', spam_c_function_add='', setup_py_add='',
```

Comment:
I agree. the subclassing is pointless, and the copy thing might be an issue down the line. hmmm. What might we call such a function..?

```python
from . import template_projects
project = template_projects.new_c_project()
```

That makes sense to me. Also, how would you feel about calling them `TestProject`s rather than `TemplateProject`s?

Reply:
>  the subclassing is pointless

I somehow always seem to fail finding a lot of decent use cases for subclassing in Python. So maybe it could make sense here. I guess we just don't now if/how this would evolve in the future. The `copy` thing is also more of a theoretical/pedantic than practical issue.

> What might we call such a function..?

Yes, `new_c_project` or `create_c_project` or ... all sounds fine.

> Cfr. the `TemplateProject.copy` from above: if this is only providing a constructor `__init__`, we might not need to subclass but just have a function creating a `TemplateProject`?

Sounds good! I guess the template is more about the implementation than about the functionality, indeed!

Reply:
> I somehow always seem to fail finding a lot of decent use cases for subclassing in Python. So maybe it could make sense here. I guess we just don't now if/how this would evolve in the future. The `copy` thing is also more of a theoretical/pedantic than practical issue.

No I think you're right, I'm not a big fan of deep class hierarchies either.

> Sounds good! I guess the template is more about the implementation than about the functionality, indeed!

I guess you're talking about `TemplateProject`/`TestProjects` here?

Reply:
> > 
> 
> I guess you're talking about `TemplateProject`/`TestProject` here?

Yes, indeed, sorry. Wrong quote.